src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -0,0 +1,249 @@
+
+/**
+ * Knows the basic conventions to run Symfony apps.
+ *
+ * Accepts two options:
+ *  - "debug" to toggle debugging features;
+ *  - "env" to define the name of the environment the app runs in.
+ *
+ * When these options are not defined, they will fallback:
+ *  - to reading the "APP_DEBUG" and "APP_ENV" environment variables;
+ *  - to parsing the "--env|-e" and "--no-debug" command line arguments
+ *    if the "symfony/console" component is installed.
+ *
+ * When the "symfony/dotenv" component is installed, .env files are loaded.
+ * When "symfony/error-handler" is installed, it is used to improve error handling.
+ *
+ * On top of the base arguments provided by the parent runtime,
+ * this runtime can feed the app-closure with arguments of type:
+ *  - Request from "symfony/http-foundation" if the component is installed;
+ *  - Application, Command, InputInterface and/or OutputInterface
+ *    from "symfony/console" if the component is installed.
+ *
+ * This runtime can handle app-closures that return instances of either:
+ *  - HttpKernelInterface,
+ *  - Response,
+ *  - Application,
+ *  - Command,
+ *  - or Closure(): int
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class SymfonyRuntime extends BaseRuntime
+{
+    private $request;
+    private $input;
+    private $output;
+    private $application;
+    private $command;
+    private $env;
+
+    public function __construct(array $options = [])
+    {
+        $this->env = $options['env'] ?? null;
+        $_SERVER['APP_ENV'] = $options['env'] ?? $_SERVER['APP_ENV'] ?? null;
+
+        parent::__construct(['debug' => $options['debug'] ?? true]);
+
+        if (isset($_SERVER['argv']) && null === $this->env && class_exists(ArgvInput::class)) {
+            $this->getInput();
+        }
+
+        if (class_exists(MissingDotenv::class, false)) {
+            $_SERVER['APP_DEBUG'] = filter_var($options['debug'] ?? $_SERVER['APP_DEBUG'] ?? true, FILTER_VALIDATE_BOOLEAN) ? '1' : '0';
+        } else {
+            (new Dotenv())->bootEnv(\dirname($_SERVER['APP_VENDOR_DIR'] ?? \dirname(__DIR__, 2)).'/.env');
+        }
+
+        if (class_exists(MissingErrorHandler::class, false)) {
+            // no-op
+        } elseif ($_SERVER['APP_DEBUG']) {
+            umask(0000);
+            Debug::enable();
+        } else {
+            ErrorHandler::register();
+        }
+    }
+
+    public function resolve(\Closure $app): \Closure
+    {
+        $app = parent::resolve($app);
+
+        return null === $this->command ? $app : $this->wrapCommand($app);
+    }
+
+    public function start(object $app): \Closure
+    {
+        if ($app instanceof HttpKernelInterface) {
+            $request = $this->request ?? $this->request = Request::createFromGlobals();
+
+            return function () use ($app, $request): int {
+                $response = $app->handle($request);
+                $status = $this->start($response)();
```

Comment:
Why "starting" again? 
IMHO, this deserve a comment in the code
(the real question is: what is the use case for supporting an instance a Response as applicatoin)

Reply:
For decoupling.

The use case is being able to write custom logic for dealing with `Response` in a loop (eg php-pm like long-running loop) and automatically be able to run a Kernel in a loop by doing so.

The opposite (hardcoding how a `Response` is handled in the logic that deals with a `Kernel`) is tight coupling and fails reusing the logic that is only for a `Kernel` instance.

(typically, long-running loops won't call `$response->send()` - calling `$this->start($response)` decouples how this these are handled.)